src/webapp_vite/vitest.setup.ts

```diff
@@ -1,9 +1,46 @@
-import { expect, afterEach } from "vitest";
+// this setup script is run before each test file.
+//
+
+import { expect, afterAll, afterEach, beforeAll } from "vitest";
 import { cleanup } from "@testing-library/react";
 import * as matchers from "@testing-library/jest-dom/matchers";
 
+import { promisify } from "node:util";
+import child_process from "node:child_process";
+const execAsync = promisify(child_process.exec);
+
+import getPort from "get-port";
+
 expect.extend(matchers);
 
+// before all test in each single test file
+beforeAll(async () => {
+    const port = await getPort();
+    const threadId = process.env.VITEST_POOL_ID;
+
+    globalThis.hubPort = port;
+    console.log("vitest.setup.ts beforeAll: starting services");
+    const { stdout, stderr } = await execAsync(
+        `cd ../.. && STRONGARM_ENV=test STRONGARM_FILE_APPEND=${threadId} HUB_PORT=${port} ./start.sh`
+    );
+    console.log(`vitest.setup.ts beforeAll start.sh stdout: ${stdout}`);
+    console.error(`vitest.setup.ts beforeAll start.sh stderr: ${stderr}`);
+});
```

Comment:
TODO in later PR,  move this code to a helper function that optionally gets called if the test file needs it, from it's own `beforeAll`  With a little be of mocking to hubMessages.ts, it would be possible to load and test every other component except `<App/>` by passing hubState data to it.